src/mbgl/style/expression/assertion.cpp

```diff
@@ -35,6 +35,23 @@ ParseResult Assertion::parse(const Convertible& value, ParsingContext& ctx) {
     return ParseResult(std::make_unique<Assertion>(it->second, std::move(parsed)));
 }
 
+const char* Assertion::getOperator() const {
+    // TODO: Should we strip assertions entirely from serialization?
+    type::Type t = getType();
+    if (t.is<type::NumberType>()) {
+        return "number";
+    } else if (t.is<type::StringType>()) {
+        return "string";
+    } else if (t.is<type::BooleanType>()) {
+        return "boolean";
+    } else if (t.is<type::ObjectType>()) {
+        return "object";
+    } else {
+        assert(false);
+        return "";
+    }
```

Comment:
This can just be:
```cpp
return type::toString(getType());
```

Reply:
Not with the `char *` interface because there wouldn't be anyone to own the string. But yeah maybe `getOperator` should just return a `std::string`...